lib/order.rb

```diff
@@ -0,0 +1,68 @@
+require 'pry'
+require 'csv'
+
+class Order
+
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status =:pending )
+
+    # check the validation of the status
+    valid_fulfillment = [:pending, :paid, :processing, :shipped, :complete]
+    raise ArgumentError.new("invalid status") unless valid_fulfillment.include?(fulfillment_status)
+
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+  end
+
+  # method to calculate the total cost of the order
+  def total
+    return  0 if @products.empty?
+    total = (@products.sum { |name, price| price } * 1.075).round(2)
+    return total
+  end
+
+  # method to add the data to the product collection
+  def add_product(product_name, price)
+    raise ArgumentError.new ("products already added") if @products.include?(product_name)
+    @products[product_name] = price
+  end
+
+
+  # helper method to create a product hash
+  # products = name:price;nextname:nextprice
+  def self.product_hash(products)
+    product = {}
+    product_arr = products.split(";")
+
+    product_arr.each do |i|
+      # [["Lobster", "17.18"], ["Annatto seed", "58.38"], ["Camomile", "83.21"]]
+      items-arr = i.split(":")
+      #assign key & value to the hash(product)
+      product[items-arr[0]] = items-arr[1].to_f
```

Comment:
So, `items-arr` is breaking the vast majority of your wave 2 tests here. I don't think your logic is incorrect, but any variable with a `-` in the name is interpreted as two variables with subtraction.

I'm also not sure why it got turned in like this when you wrote complete tests?